### PR TITLE
kill subscriber process when desktop is removed

### PR DESCRIPTION
**Pull request by no3m**
change the way kill_layout gets called... desktop options in remove_listener have to be cleared before calling kill_layout, otherwise if remove_listener is called by the subscriber process on desktop_remove event, it will kill itself before the options are cleared.

added 'did' (desktop id) option... once desktop_remove event is triggered, get_desktop_name_from_id is useless since desktop doesn't exist anymore.

### CODE REVIEW

File: src/layout.sh [hunk 1]
```diff
@@ -132,11 +143,14 @@ start_listener() {
   disown;
 
   # Kill old layout
-  kill_layout $selected_desktop;
+  local old_pid="$(get_desktop_options "$selected_desktop" | valueof pid)";
+  kill_layout $old_pid;
 
+  desktop_id=$(bspc query -D -d "$selected_desktop");
```
Comment (phenax) [inline src/layout.sh:149]: nit: this can be a function inside desktop.sh

File: src/layout.sh [hunk 2]
```diff
@@ -14,19 +14,22 @@ BSP_DEFAULT_LAYOUTS="tiled\nmonocle";
 
 # Kill old layout process
 kill_layout() {
```
Comment (phenax) [inline src/layout.sh:16]: `kill_layout`'s purpose is to kill the layout of a desktop. It doesn't make sense for it to accept a pid. Maybe we can have 2 functions `kill_layout` and `kill_process`

**Review by phenax — CHANGES_REQUESTED** [review]
Sorry for the delay. There's some changes needed here. Let me know if you'll be able to make those changes, if not I'll pick this up when I can and merge it in.

**Comment by a2n-s** [comment]
@phenax, in case @no3m does not respond, would you like me to try to fix the requests you made on the commit? :yum:

**Comment by a2n-s** [comment]
Hi @no3m :wave: :yum: 

as this PR has ben stale for almost 8 months now, we've decided with @phenax to move the changes to [a2n-s/kill-subscriber-process-when-desktop-is-removed](https://github.com/a2n-s/bsp-layout/tree/kill-subscriber-process-when-desktop-is-removed) and apply the requested changes there :+1:

hope you'll understand :wink:

**Pull request closed without merging**